packages/utils/tool-utils/src/odspTokenManager.ts

```diff
@@ -50,12 +50,27 @@ export type OdspTokenConfig = {
 
 export interface IOdspTokenManagerCacheKey { isPush: boolean; server: string; }
 
+const isValidToken = (token: string) => {
+    const decodedToken = jwtDecode<any>(token);
+    // Give it a 60s buffer
+    return (decodedToken.exp - 60 >= (new Date().getTime() / 1000));
+};
+
+const cacheKeyToString = (key: IOdspTokenManagerCacheKey) => {
+    return `${key.server}${key.isPush ? "[Push]" : ""}`;
+};
+
 export class OdspTokenManager {
+    private readonly storageCache = new Map<string, IOdspTokens>();
+    private readonly pushCache = new Map<string, IOdspTokens>();
```

Comment:
cacheKeyToString  takes in account isPush, so why do we need two caches?

Reply:
Because Map key doesn't do deep object equality check?